src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -972,6 +976,30 @@ private void AssignUnscheduledRequestsByTraversalsFirst(List<ScheduleResponse> r
             }
         }
 
+        /// <summary>
+        /// Proxy build requests <see cref="ProxyTargets"/> should be really cheap (only return properties and items) and it's not worth
+        /// paying the IPC cost and re-evaluating them on out of proc nodes (they are guaranteed to be evaluated in the Scheduler process).
+        /// </summary>
+        private void AssignUnscheduledProxyBuildRequestsToInProcNode(List<ScheduleResponse> responses, HashSet<int> idleNodes)
+        {
+            if (idleNodes.Contains(InProcNodeId))
+            {
+                List<SchedulableRequest> unscheduledRequests = new List<SchedulableRequest>(_schedulingData.UnscheduledRequestsWhichCanBeScheduled);
```

Comment:
Why do you need to wrap this in a new list? UnscheduledRequestsWhichCanBeScheduled is already IEnumerable

Reply:
I was just copy pasting the existing patterns of accessing those requests. I suppose it's for concurrency purposes. The method needs a snapshot, not a live list.